### DISCUSSION TITLE
ABL before each print stopped working after the latest firmware update (E3V2 4.2.2 2/ CR Touch)

### DISCUSSION THREAD

**Discussion by czy5197** [mriscoc/Ender3V2S1#584] [Q&A]
I just upgraded to the latest firmware but somehow ABL stopped working. Before I just used G29 to enable a bed leveling prior to each print, after the upgrade the printer just directly goes to printing after homing, skipping the ABL.

I looked up the guide in this repo and added G29 A to enable probing (for some reason Marlin's official guide says G28 disables ABL), and then added M420 S1 Z2 as the guide here specified. Still nothing. I made sure to recompile the G code and upload it to the printer. The ABL function itself works when I manually select that function on the home screen. Also the double underline under Z symbol isn't blue, according to the guide, this means mesh compensation is not enabled. Anyone knows what's going on here?

**Comment by mriscoc** [comment]
Look at the release page, now the firmware uses UBL as the default leveling system.
